node/subsystem-util/src/validator_discovery.rs

```diff
@@ -48,6 +48,48 @@ pub enum Error {
 	RuntimeApi(#[from] RuntimeApiError),
 }
 
+/// Utility function to connect to different sets of validators at different relay parents.
+pub async fn connect_to_validators_at_different_heights<Context: SubsystemContext>(
+	ctx: &mut Context,
+	// (relay_parent, validators)
+	validators: Vec<(Hash, Vec<ValidatorId>)>,
+) -> Result<ConnectionRequest, Error> {
+	let mut validator_map = HashMap::new();
+	let mut authorities = Vec::new();
+
+	for (relay_parent, validators) in validators {
+		// ValidatorId -> AuthorityDiscoveryId
+		let (tx, rx) = oneshot::channel();
+
+		ctx.send_message(AllMessages::RuntimeApi(
+			RuntimeApiMessage::Request(
+				relay_parent,
+				RuntimeApiRequest::ValidatorDiscovery(validators.clone(), tx),
+			)
+		)).await?;
+
+		let maybe_authorities = rx.await??;
+
+		authorities.extend(maybe_authorities.iter()
+			.cloned()
+			.filter_map(|id| id)
```

Comment:
Unless I'm mistaken, this line doesn't do anything.

```suggestion
```

Reply:
[This filters `Option<B>` into `B`.](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=181168f83ca76677f07d31bf1b0fb605)